### PR TITLE
House Cleaning - Part 1 [Ongoing]

### PR DESCRIPTION
**Pull request by lzardy**
- Adds comments to many classes, their functions, and variables and reorganizes them (mostly alphabetically).
- Added a temporary workaround for Windows incompatibility by editing the x86 target platform. Edited the README for new build instructions.
- Cleaned some functions in PacketNotifier, ApiFunctionManager, Champion, LaneMinion, Minion, CollisionHandler, and Spell.
- Added some implementations of the LeaguePackets packets in PacketNotifier and replaced some old LS packet implementations.
- Added some mathematical functions in Extensions which could be used for to make a dynamic NavigationGrid.

Due to the monolithic nature of commenting and cleaning everything, future "House Cleaning" PRs will be made as continuations so as to not overwhelm the reviewing process and to be sure each part is up to quality.

### CODE REVIEW

File: GameServerLib/API/ApiFunctionManager.cs [hunk 1]
```diff
@@ -74,25 +120,52 @@ public static void SetGameObjectVisibility(GameObject gameObject, bool visibilit
...
+        /// <param name="x">X coordinate.</param>
+        /// <param name="y">Y coordinate.</param>
         public static void TeleportTo(IObjAiBase unit, float x, float y)
         {
-            var coords = new Vector2(x, y);
-            var truePos = _game.Map.NavigationGrid.GetClosestTerrainExit(coords);
-
             CancelDash(unit);
-            unit.TeleportTo(truePos.X, truePos.Y);
+            unit.TeleportTo(x, y);
```
Comment (danil179) [inline GameServerLib/API/ApiFunctionManager.cs:143]: What happens if you teleport into a wall?

File: GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs [hunk 2]
```diff
@@ -93,6 +86,11 @@ public override void Update(float diff)
 
         public virtual bool AIMove()
         {
+            if (IsLaneMinion)
+            {
+                (this as ILaneMinion).WalkToDestination();
```
Comment (danil179) [inline GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs]: This is not quite the same as the previous version. The previous version invokes the code below and then conditioning the if here based on that. What is the reason for this change?

As a general principle, if you need to use down-casting like here, then doing virtual functions is better (the old solution).

File: GameServerLib/GameObjects/GameObject.cs [hunk 3]
```diff
@@ -80,25 +99,136 @@ public GameObject(Game game, float x, float y, int collisionRadius, int visionRa
...
+        {
+            X = vec.X;
+            Y = vec.Y;
+            Target = null;
+        }
+
+        /// <summary>
+        /// Refers to the height that the object is at in 3D space. *NOTE*: Should be renamed.
+        /// </summary>
+        // TODO: Change this to property to GetY() as it does not actually represent the Z-axis in 3D space, it currently represents the Y-axis
```
Comment (danil179) [inline GameServerLib/GameObjects/GameObject.cs]: Probably better stated as `GetHeight` as X,Y,Z are really depends on the orientation.

**Review by danil179 — APPROVED** [review]
Thank you very much for adding such comprehensive explanations and documentation!

File: GameServerLib/API/ApiFunctionManager.cs (hunk 1 above)
Comment (lzardy) [reply GameServerLib/API/ApiFunctionManager.cs:143]: This calls ObjAiBase.TeleportTo, which already gets the closest terrain exit, I just removed it because it would be a useless extra call.

File: GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs (hunk 2 above)
Comment (lzardy) [reply GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs]: Forgot about the virtual, and while I'm at it I'll just re-add override AIMove since I can achieve the same thing with it. My intention is to separate the Minion AI functions from the LaneMinion AI functions; Minion.ScanForTargets/KeepFocusingTarget shouldn't be used for LaneMinions.

**Review by danil179 — APPROVED** [review]
Approved. I hope it all works well (specifically lane minions attacks).

**Comment by lzardy** [comment]
LaneMinions shouldn't attack because their AI isn't fully implemented, it was intentional. I have an AI rework branch I plan to implement in the future and figured fully separating Minion from LaneMinion was a good change for a "cleanup" PR.

**Comment by danil179** [comment]
Refs #1029

File: Content/LeagueSandbox-Scripts/LeagueSandbox-Scripts.csproj [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>netcoreapp3.0</TargetFramework>
-    <Platforms>AnyCPU;x64</Platforms>
+    <Platforms>AnyCPU;x86</Platforms>
```
Comment (MythicManiac) [inline Content/LeagueSandbox-Scripts/LeagueSandbox-Scripts.csproj:5]: ❓

Comment (lzardy) [reply Content/LeagueSandbox-Scripts/LeagueSandbox-Scripts.csproj:5]: x86 was the only platform GameServerConsole would build with on Windows as far as I tested.

Comment (MythicManiac) [reply Content/LeagueSandbox-Scripts/LeagueSandbox-Scripts.csproj:5]: This only makes me wonder what was preventing GameServerConsole from building on other architectures, there shouldn't be anything that makes it required

Comment (lzardy) [reply Content/LeagueSandbox-Scripts/LeagueSandbox-Scripts.csproj:5]: I believe it was a DLL issue with ENet, however even when I tested x32 and x64 built DLLs (from the ENet repo.), it did not resolve the issue.

Comment (lzardy) [reply Content/LeagueSandbox-Scripts/LeagueSandbox-Scripts.csproj:5]: Building on AnyCPU on Windows should give you the exact error.

Comment (MythicManiac) [reply Content/LeagueSandbox-Scripts/LeagueSandbox-Scripts.csproj:5]: GameServerConsole shouldn't have a dependency to ENet, right?

Comment (MythicManiac) [reply Content/LeagueSandbox-Scripts/LeagueSandbox-Scripts.csproj:5]: Interestingly enough I was sure x64 was the only build shipped with nuget, I might be wrong though. The repo for that is here: https://github.com/LeagueSandbox/ENetSharpLeague and build config here: https://github.com/LeagueSandbox/ENetSharpLeague/blob/indev/ENetCS/ENetCS.csproj

Comment (lzardy) [reply Content/LeagueSandbox-Scripts/LeagueSandbox-Scripts.csproj:5]: Honestly no idea, It runs fine with x86, but not x64, x32, or Any CPU. The problem is with GameServerConsole in particular. Setting its platform target to x86 for Any CPU also fixes it, but I didn't want to do this as it seems to break the build for Linux users. Hence the addition of a fully x86 build for Windows users.
